CMakeLists.txt

```diff
@@ -7,6 +7,8 @@ if(NOT CMAKE_BUILD_TYPE)
       "Choose the type of build, options are: Debug Release RelWithDebInfo." FORCE)
 endif()
 
+set(CMAKE_LINK_WHAT_YOU_USE TRUE)
```

Comment:
https://cmake.org/cmake/help/v3.7/prop_tgt/LINK_WHAT_YOU_USE.html#prop_tgt:LINK_WHAT_YOU_USE
Needed for ubuntu builds, due to different linking policy used by system's gcc